apps/coscrad-frontend/src/components/resources/digital-text/digital-text-detail.full-view.presenter.tsx

```diff
@@ -1,19 +1,151 @@
+                        dispatch(
+                            executeCommand({
+                                type: 'ADD_CONTENT_TO_DIGITAL_TEXT_PAGE',
+                                payload: {
+                                    aggregateCompositeIdentifier: {
+                                        type: AggregateType.digitalText,
+                                        id,
+                                    },
+                                    text,
+                                    languageCode,
+                                    pageIdentifier,
+                                },
+                            })
+                        )
+                    }
+                />
+            ) : null}
+            <Button
+                onClick={() => {
+                    setCurrentIndex(cyclicDecrement(currentIndex, pages.length));
+                }}
+            >
+                {'<< PREV'}
+            </Button>{' '}
+            <Button
+                onClick={() => {
+                    setCurrentIndex(cyclicIncrement(currentIndex, pages.length));
+                }}
+            >
+                {'NEXT >>'}
+            </Button>
+            {/* TODO I am a back-end developer */}
+            <br />
+            <NewPageForm
+                existingPageIdentifiers={allExistingPageIdentifiers}
+                onSubmitPageIdentifier={(pageIdentifier) => {
+                    dispatch(
+                        executeCommand({
+                            type: 'ADD_PAGE_TO_DIGITAL_TEXT',
+                            payload: {
+                                aggregateCompositeIdentifier,
+                                identifier: pageIdentifier,
+                            },
+                        })
+                    );
+                }}
+            />
+            {!isPublished ? (
+                <PublicationForm
+                    onSubmitForPublication={() => {
+                        dispatch(
+                            executeCommand({
+                                type: 'PUBLISH_RESOURCE',
+                                payload: {
+                                    aggregateCompositeIdentifier: aggregateCompositeIdentifier,
+                                },
+                            })
+                        );
+                    }}
+                />
+            ) : (
+                <Typography variant="body2">Published</Typography>
+            )}
+            <br />
+            {/* TODO Include tags */}
+            Tags:{' '}
+            <CommaSeparatedList>
+                {tags.map(({ label }) => (
+                    <div>{label}</div>
+                ))}
+            </CommaSeparatedList>
+            <br />
+            <ImmersiveCreateNoteForm
+                onSubmit={(text, languageCode, noteId) => {
+                    dispatch(
+                        executeCommand({
```

Comment:
It would be nice if we didn't have to repeat this callback definition every time we want to leverage this form.